### ISSUE TITLE
[Core] Turn on WAL mode for cluster job table

### ISSUE THREAD

**Issue by Michaelvll** [skypilot-org/skypilot#3863]
For job table on clusters, we do not turn on `WAL` mode, which causes errors when user cancels and submits new jobs quickly and frequently:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/root/skypilot-runtime/lib/python3.10/site-packages/sky/skylet/log_lib.py", line 448, in tail_logs
    for line in _follow_job_logs(log_file,
  File "/root/skypilot-runtime/lib/python3.10/site-packages/sky/skylet/log_lib.py", line 381, in _follow_job_logs
    status = job_lib.get_status_no_lock(job_id)
  File "/root/skypilot-runtime/lib/python3.10/site-packages/sky/skylet/job_lib.py", line 339, in get_status_no_lock
    rows = _CURSOR.execute('SELECT status FROM jobs WHERE job_id=(?)',
sqlite3.OperationalError: database is locked
```

**Comment by wizenheimer** [comment]
Hey @Michaelvll,
Pushed a fix. Please take a [look](https://github.com/skypilot-org/skypilot/pull/3923).

**Comment by Michaelvll** [comment]
This should be fixed by #3923. Closing
